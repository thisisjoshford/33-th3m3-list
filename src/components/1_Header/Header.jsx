import React from 'react';
import Toggle from '../ThemeToggle/Toggle';
const Header = () => {
  return (
    <>
      <h1>X-Files SEARCH API</h1>
      <Toggle />
    </>
  );
};

export default Header;
