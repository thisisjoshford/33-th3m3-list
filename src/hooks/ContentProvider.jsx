import React, { createContext, useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { getPeople } from '../services/apiFetch';

export const XfilesContent = createContext();

export const ContentProvider = ({ children }) => {
  const [characters, setCharacters] = useState([]);

  useEffect(() => {
    getPeople()
      .then(peeps => setCharacters(peeps));
  }, []);

  return (
    <XfilesContent.Provider value={characters}>
      {children}
    </XfilesContent.Provider>
  );
};

ContentProvider.propTypes = {
  children: PropTypes.node
};

