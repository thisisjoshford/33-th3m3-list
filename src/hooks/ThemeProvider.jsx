import React, { createContext, useState, useContext } from 'react';
import PropTypes from 'prop-types';

const ThemeContext = createContext();

export const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState('light');
  const toggle = ({ target }) => {
    if(target.checked) setTheme('dark');
    if(!target.checked) setTheme('light');
  };

  return (
    <ThemeContext.Provider value={{ theme, toggle }}>
      {children}
    </ThemeContext.Provider>
  );
};

ThemeProvider.propTypes = {
  children: PropTypes.node
};

export const useTheme = () => {
  const theme = useContext(ThemeContext);
  return theme;
};

export const useToggle = () => {
  const toggle = useContext(ThemeContext);
  return toggle;
};
