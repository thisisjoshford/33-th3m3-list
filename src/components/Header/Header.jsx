import React from 'react';
import Toggle from '../ThemeToggle/Toggle';
import { useTheme, useToggle } from '../../hooks/ThemeProvider';
import Styling from '../ListPage/ListPage.css';

const Header = () => {
  const { theme } = useTheme();
  const { toggle } = useToggle();

  let styling;
  if(theme === 'light') styling = Styling.Light;
  if(theme === 'dark') styling = Styling.Dark;

  return (
    <>
      <h1 className={styling}>X-Files SEARCH API</h1>
      <Toggle value={theme === 'dark'} toggle={toggle} />
    </>
  );
};

export default Header;
