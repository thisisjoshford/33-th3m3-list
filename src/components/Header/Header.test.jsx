import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';
import { ThemeProvider } from '../../hooks/ThemeProvider';

describe('Header component', () => {

  it('renders Header', () => {
    const wrapper = shallow(
      <ThemeProvider>
        <Header/>
      </ThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
