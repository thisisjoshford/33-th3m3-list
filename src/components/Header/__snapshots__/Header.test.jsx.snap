// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header 1`] = `
<ContextProvider
  value={
    Object {
      "theme": "light",
      "toggle": [Function],
    }
  }
>
  <Header />
</ContextProvider>
`;
